haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/GhcApiSupport.hs

```diff
@@ -229,6 +231,18 @@ withTypechecked globalLock GhcArgs{..} analysisOpts xrefSink
             -- This might be needed if TH executes code from other package? Or
             -- only if that code needs FFI?
 
+-- | Doc decls are for things that don't have source code available; most
+-- possibly those from core packages. Hence their decls may be duplicated
+-- coming from multiple analyses. Depuplicate them, and put them in the
+-- 'xrefDecls' field of the first 'XRef'.
+postProcessDocDecls :: [XRef] -> [XRef]
+postProcessDocDecls graph =
+    let docDecls = concatMap xrefDocDecls graph
+        deduped = S.toList $ foldr S.insert S.empty docDecls
+     in case graph of
+            [] -> []
+            x : xs -> x {xrefDecls = xrefDecls x ++ deduped} : xs
```

Comment:
Does this correctly dedup elements beyond the first one?  (I.e., those in `xs`.)  Is that a concern?

Reply:
Dedup doesn't happen onto `graph` but onto `docDecls`. The `xrefDocDecls` fields are not used after this phase. Only `xrefDecls` fields are used. That's why I'm appending `deduped` to the `xrefDecls` field of the first item, rather than to the `xrefDocDecls` field.

To clarify the intention, now I'm clearing all the `xrefDocDecls` fields here.